### PR TITLE
[AAP-21457] Add workflow nodes output toolbar status bar

### PR DESCRIPTION
**Pull request by jerabekjiri**
Issue: [AAP-21457](https://issues.redhat.com/browse/AAP-21457)

- Generalize `HostStatusBar` into a reusable `StatusBar` component.
- Create `HostStatusBar` and `WorkflowNodesStatusBar` components based on the `StatusBar` component.

Screenshot from 2024-03-13 23-49-12

Screenshot from 2024-03-13 22-15-57

Screenshot from 2024-03-14 22-22-37

### CODE REVIEW

File: frontend/awx/views/jobs/JobOutput/StatusBar.tsx [hunk 1]
```diff
@@ -47,17 +48,68 @@ const LegendBox = styled.span`
...
+    pending: {
+      color: pfInfo,
+      label: t`Pending`,
+    },
+    error: {
+      color: pfDanger,
+      label: t`Failed`,
+    },
+    unreachable: {
+      color: 'var(--pf-v5-global--danger-color--300)',
```
Comment (nixocio) [inline frontend/awx/views/jobs/JobOutput/StatusBar.tsx:91]: What about create a pfUnreachable color?

File: frontend/awx/views/jobs/JobOutput/StatusBar.tsx [hunk 2]
```diff
@@ -81,27 +133,44 @@ export function HostStatusBar(props: { counts: HostStatusCounts }) {
...
+  const { counts, status } = props;
+  const noData = Object.keys(counts).length === 0;
+  const totalCounts = Object.values(counts).reduce(
+    (sum: number, count: number) => sum + count,
+    0
+  ) as number;
+
+  const barSegments = Object.keys(status).map((key) => {
+    const count = (counts[key as keyof T] as number) || 0;
+    const jobStatus = status[key as keyof K] as StatusProps;
```
Comment (nixocio) [inline frontend/awx/views/jobs/JobOutput/StatusBar.tsx:155]: I am getting an error locally here.

image

It seems like we could add default values to the color and label.

**Review by nixocio — CHANGES_REQUESTED** [review]

File: frontend/awx/views/jobs/JobOutput/StatusBar.tsx (hunk 1 above)
Comment (jerabekjiri) [reply frontend/awx/views/jobs/JobOutput/StatusBar.tsx]: done :)

**Review by resoluteCoder — APPROVED** [review]
lgtm

**Review by nixocio — APPROVED** [review]

**Pull request merged**